### DISCUSSION TITLE
WebSocket showing closed for Application deployed with Istio Service Mesh and Ingress Gateway

### DISCUSSION THREAD

**Discussion by kr-swarna** [istio/istio#50885] [Q&A]
**Prerequisite:** Helm charts such as istio-base, istio-discovery, istio-ingress-gateway with app version 1.0.0 applied to the namespace before applying the following application.

I have deployed the 'Tornado - Demo Websockets App' from the link below:
[https://github.com/istio/istio/tree/master/samples/websockets](url)

in the AWS Elastic Kubernetes Cluster. I labeled the namespace with **'istio-injection=enabled'** and modified the deployment in the app.yaml file by adding the annotation below:
`annotations:
    sidecar.istio.io/inject: "true" 
`

However, the WebSocket status remains closed when the application is up.
---------------------

What should be done to make the WebSocket status open?

**Comment by zirain** [comment]
you should use lates version of istio, which means `1.21.2`.

**Reply by kr-swarna** [reply]
I see that the istio discovery pod is running with the latest image,
`Image:         gcr.io/istio-testing/pilot:latest`
`Image ID:      gcr.io/istio-testing/pilot@sha256:2a974ef2bd3cdca71dc2cd9b2b463f8441b70e77ecbd9e05f7f5200cd81e3979`
    
and the deployment is running with the below image,
`Image:         gcr.io/istio-testing/proxyv2:latest`
`Image ID:      gcr.io/istio-testing/proxyv2@sha256:8c358d37d33c013fa5dc8b1c331030e19383a8d02b83c90e27d8d47e4209622e`

**Reply by howardjohn** [reply]
**DO NOT USE LATEST**. Please follow https://istio.io/latest/docs/setup/install/helm/. Probably you were installing from source which is not supported.

**Reply by K-R-Swarna** [reply]
![image](https://github.com/istio/istio/assets/74405227/d653faed-02d0-4477-9323-97f79607b437)
![image](https://github.com/istio/istio/assets/74405227/6a0a1d01-9ef7-40e0-9bf7-89718ea52804)

NAME     READY   UP-TO-DATE   AVAILABLE   AGE   CONTAINERS   IMAGES                         SELECTOR
istiod   1/1     1            1           23d   discovery    docker.io/istio/pilot:1.21.2   istio=pilot

this is the version we have currently deployed for istio and the issue is with respect to a pod which uses SignalR and it is showing as WebSocket closed, how to resolve this?
